### ISSUE TITLE
Render Cucumber "features" as articles

### ISSUE THREAD

**Issue by plexus** [cljdoc/cljdoc#257]
Cucumber is a multi-platform testing tool, when using Cucumber you write your tests in a language called "Gherkin", which looks like lightly structured plain text. Each test file is known as a "feature", and uses the `.feature` extension (these are meant to be "integration" tests, so supposedly each file describes a single feature of your library/app).

Here's an [example of a feature file](https://github.com/lambdaisland/kaocha/blob/master/test/features/pending.feature). This can be executed as a test (provided you provide definitions of the "steps" used, [here's an example of that](https://github.com/lambdaisland/kaocha/blob/master/test/step_definitions/kaocha_integration.clj#L172-L174)). 

The benefit of writing your tests in this roundabout way is that your tests are now optimized for being read by humans, and so can straightforwardly be rendered as HTML or Markdown. Here's the same "test" [rendered as a cljdoc article](https://cljdoc.org/d/lambdaisland/kaocha/0.0-359/doc/marking-tests-as-pending).

To try this out I've generated the markdown and checked it into the repo, but now I have generated files checked in that I need to manually keep up to date with the source files. If cljdoc could understand `.feature` files the way it understands `.md` or `.adoc` files this would be much more straightforward, and allow for nicer rendering of the output, generating a table of contents, etc.

If this is considered useful I'd be happy to contribute support for it. I already did the work of wrapping `cucumber-jvm`, so it's easy to parse a `.feature` file to a plain EDN data structure.

I'm not 100% sure yet what the behavior should be regarding detecting `.feature` files, and combining them with other articles, but I can follow up with a more concrete proposal to discuss.

Note that this could be considered related to #70, but it's not really the same thing, so I'd rather not pull that whole discussion into this one. It could be a good starting point though, e.g. with this in place we could think about ways to make examples in Cucumber tests interactive.

**Comment by plexus** [comment]
That's right, but it's no different from how articles are currently rendered.

**Comment by plexus** [comment]
The easiest first step would be to make it opt-in, by referencing each feature file explicitly from `cljdoc.edn`

``` edn
{:cljdoc.doc/tree
 [["Do this thing" {:file "test/features/do_this_thing.feature"}]]}
```

That seems like a fairly uncontroversial starting point that should be easy to implement.

Projects that really embrace Cucumber as documentation on the other hand will have a structure like this

```
features
├── README.md
├── command_line_args
│   ├── README.md
│   ├── no_color.feature
│   └── fail_fast.feature
└── plugins
    ├── README.md
    ├── version_filter.feature
    └── capture_output.feature
```

Here the idea is that each directory gets rendered as `README.md` followed by a table of contents of the next level. This is what RSpec does, and what I'd like to do with Kaocha. Compare [source](https://github.com/rspec/rspec-core/tree/master/features) and [rendered result](https://relishapp.com/rspec/rspec-core/v/3-8/docs).

Maybe something like this?

``` clojure
{:cljdoc.doc/feature-paths ["test/features"]}
```

Finally, should cljdoc look for `*.feature` files automatically? e.g. in `features` or `test/features`? I personally think this would be a good thing, as it may help people discover Cljdoc's support for this, and makes docs richer by default.

**Comment by martinklepsch** [comment]
I'm doing some work to make the doctree a bit more extensible (different types of items) which is probably useful for this too. Will share a PR once I have something but just wanted to let you know so we don't do the same stuff :)

**Comment by martinklepsch** [comment]
Ok, here's some WIP: https://github.com/cljdoc/cljdoc/compare/extensible-article-types?expand=1
